src/types/ComponentType.ts

```diff
@@ -39,8 +40,20 @@ export class ComponentType extends InheritableType {
 
     public static instance = new ComponentType('Node');
 
-    isEqual(targetType: BscType): boolean {
-        return isComponentType(targetType) && this.name.toLowerCase() === targetType.name.toLowerCase();
+    isEqual(targetType: BscType, data: TypeCompatibilityData = {}): boolean {
+        if (isReferenceType(targetType) && targetType.isResolvable()) {
+            targetType = targetType.getTarget();
+        }
+        if (this === targetType) {
+            return true;
+        }
+
+        return isComponentType(targetType) && this.name.toLowerCase() === targetType.name.toLowerCase() &&
```

Comment:
A better job of checking equality of Component Types, so that we can check if a change was substantial enough to re-validate